DataImport.Web/Features/ApiServers/RefreshDataModel.cs

```diff
@@ -0,0 +1,80 @@
+using DataImport.Models;
+using DataImport.Web.Helpers;
+using DataImport.Web.Services;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace DataImport.Web.Features.ApiServers
+{
+    public class RefreshDataModel
+    {
+        public class Command : IRequest<Response>
+        {
+            public int ApiServerId { get; set; }
+        }
+
+        public class Response : ToastResponse
+        {
+            public int ApiServerId { get; set; }
+        }
+
+        public class CommandHandler : IRequestHandler<Command, Response>
+        {
+            private readonly DataImportDbContext _database;
+            private readonly IConfigurationService _configurationService;
+
+            public CommandHandler(DataImportDbContext database, IConfigurationService configurationService)
+            {
+                _database = database;
+                _configurationService = configurationService;
+            }
+
+            public async Task<Response> Handle(Command request, CancellationToken cancellationToken)
+            {
+                var apiServer = await _database.ApiServers
+                    .Include(x => x.ApiVersion)
+                    .SingleOrDefaultAsync(x => x.Id == request.ApiServerId, cancellationToken);
+
+                if (apiServer is null)
+                {
+                    throw new System.ArgumentException("Api server not found");
+                }
+
+                await _configurationService.FillSwaggerMetadata(apiServer);
+
+                await _database.SaveChangesAsync(cancellationToken);
+
+                var dataMaps = await (from dm in _database.DataMaps
+                                      join res in _database.Resources on dm.ApiVersionId equals res.ApiVersionId
+                                      where dm.ApiVersion == apiServer.ApiVersion && dm.ResourcePath.Trim() == res.Path.Trim()
+                                      select dm
+                                     ).ToListAsync(cancellationToken);
+
+                var resourcesPath = dataMaps.Select(map => map.ResourcePath).Distinct().ToList();
+
+                var metadataFromResources = await _database
+                    .Resources
+                    .Where(resource => resource.ApiVersionId == apiServer.ApiVersionId && resourcesPath.Contains(resource.Path))
+                    .Select(resource => new { resource.Path, resource.Metadata })
+                    .ToListAsync(cancellationToken);
+
+                dataMaps.ForEach(map =>
+                {
+                    map.Metadata = metadataFromResources.Where(x => x.Path == map.ResourcePath).Select(x => x.Metadata).First();
```

Comment:
@beto-devnet .First() will throw an exception if there is more than one, is that appropriate here?

Reply:
in this case, the issue could occur if the First() method returns an empty string, in that case, the Metadata should not be updated.

I modified the code and added a condition not to be updated if the string is empty